lib/chef/provider/package/yum.rb

```diff
@@ -1081,22 +1082,24 @@ def load_current_resource
                 @new_resource.version($2)
               end
             end
-          end
-
-          if @new_resource.version
-            new_resource = "#{@new_resource.package_name}-#{@new_resource.version}#{yum_arch}"
+            @candidate_version << @new_resource.version
+            installed_version << @yum.installed_version(@current_resource.package_name, arch)
```

Comment:
that will enforce source is a string, but it will not enforce we never use a source with an array of packages. FOr example, one could currently do:

```
package ['lspci', 'iptstate'] do
  source '/tmp/lspci.rpm'
end
```

Wtf does that mean? If source is present, we should ensure that package_name and version are both strings.